packages/dds/tree/src/feature-libraries/node-key/nodeKeyManager.ts

```diff
@@ -33,6 +34,18 @@ export interface NodeKeyManager {
 	 * Convert the given {@link LocalNodeKey} into its {@link StableNodeKey} form.
 	 */
 	stabilizeNodeKey(key: LocalNodeKey): StableNodeKey;
+
+	/**
+	 * Attempts to recompress a StableNodeKey.
+	 * @param key - The UUID that is attempted to recompress.
+	 * @returns The `LocalNodeKey` associated with `key` or undefined if the UUID was not generated by any session known to this compressor.
+	 */
+	tryLocalizeNodeKey(key: string): LocalNodeKey | undefined;
+
+	/**
+	 * type guard to check if a key is a {@link StableNodeKey}
+	 */
+	isStableNodeKey(key: string): key is StableNodeKey;
```

Comment:
I see two options for this function:
1. It is a free function (not an interface member) and all it does is call `isStableId()`. You want it to be a free function, not an interface member, because otherwise every implementation of the interface has to implement it the same way - it is needless duplication.
2. It is an interface member, but it gives a stronger assertion. It ensures that the given key is a `StableNodeKey` but _also_ that it _belongs_ to this `NodeKeyManager`. This is because there's a scenario that I forgot about when we last talked: you might have a `StableNodeKey` from another app, or another session (or maybe you even have two Fluid containers in a single app, each with a shared tree, and you are mixing the StableNodeKeys from each together). In that case, you want to check that the `StableNodeKey` passed here is one of "your" `StableNodeKeys`, otherwise you can'd do anything with it.

I think it probably makes sense to just do both. Have a free function `isStableNodeKey(string)` and a method on `NodeKeyManager` called `hasKey(key: StableNodeKey)`. The implementation of the latter is simply `compressor.tryRecompress(key) !== undefined`.

Reply:
Removed method from the interface and created free function `isStableNodeKey`